falcon/response.py

```diff
@@ -31,10 +32,13 @@
 from falcon.util import structures
 from falcon.util import TimezoneGMT
 from falcon.util.deprecation import deprecated
-from falcon.util.uri import encode as uri_encode
-from falcon.util.uri import encode_value as uri_encode_value
+from falcon.util.uri import encode
+from falcon.util.uri import encode_value
 
 
+uri_encode = partial(encode, check_is_escaped=True)
```

Comment:
We're a bit hardcore when it comes to performance of the key parts of the framework, and IIRC it is more performant to simply call the function with the desired _kwargs_ than to incur function call overhead via `functools.partial`.

Reply:
Ah got it, good to know. Will make the change

Reply:
Just to illustrate I'm not making this up (CPython 3.8) :slightly_smiling_face: 

```python
>>> def my_func(param, futile_flag=False):
...     return param.upper()
... 
>>> my_func('hello!')
'HELLO!'
>>> another_func = functools.partial(my_func, futile_flag=True)
>>> timeit.timeit(lambda: my_func('hello', futile_flag=True))
0.13354026100000738
>>> timeit.timeit(lambda: my_func('hello', futile_flag=True))
0.13823660299999574
>>> timeit.timeit(lambda: my_func('hello', futile_flag=True))
0.13869832399996085
>>> 
>>> timeit.timeit(lambda: another_func('hello'))
0.2074820359999876
>>> timeit.timeit(lambda: another_func('hello'))
0.2078785850000031
>>> timeit.timeit(lambda: another_func('hello'))
0.20846412799994596
```

Reply:
No way.....is that a 50% difference?? That's wild. I had no idea. Will try to fix this today.

Reply:
Was just about to push something but noticed that header_property (for content_location in response.py::913) takes a "transformer" method as an argument and then applies a value to that method. I was wondering if in this case it might be ok to use the partial?
    
Alternatively, I was originally thinking of exposing 2 more methods on the URI as encode_check_escaped and encode_check_escaped_value and then just using those in response.

Reply:
I'd like to try moving `check_is_escaped` to `_create_str_encoder()` and letting the closure do the work more efficiently as compared to `partial()`.  Having two additional methods would be fine... we can also let people know (in the news fragment) that if they were using `encode()` and `encode_value()` directly, they can work around the breaking change by using the new additional methods.

That being said, I would prefer `encode_value_check_escaped()` in lieu of `encode_check_escaped_value()`. There may also be a more elegant way to say it than "check escaped" if we give it a little more thought.

Thanks for tackling this!

Reply:
Ok, great, I'll make that change today.